src/network/network_client.cpp

```diff
@@ -1171,9 +1171,9 @@ void ClientNetworkGameSocketHandler::CheckConnection()
 	/* 20 seconds are (way) more than 4 game days after which
 	 * the server will forcefully disconnect you. */
 	if (lag > 20) {
+		ClientNetworkEmergencySave();
```

Comment:
as CloseConnection() also save and show the error message, maybe just remove them here